v0/destinations/yahoo_dsp/transform.js

```diff
@@ -0,0 +1,125 @@
+const sha256 = require("sha256");
+const { BASE_ENDPOINT, ENDPOINTS } = require("./config");
+const {
+  defaultRequestConfig,
+  CustomError,
+  isDefinedAndNotNullAndNotEmpty,
+  defaultPutRequestConfig,
+  removeUndefinedAndNullValues
+} = require("../../util");
+
+const populateIdentifiers = (attributeArray, { Config }) => {
+  const seedList = [];
+  const { audienceType } = Config;
+  const { hashRequired } = Config;
+  let listType;
+  const AUDIENCE_LIST = ["email", "deviceId", "ipAddress"];
+  if (isDefinedAndNotNullAndNotEmpty(attributeArray)) {
+    // traversing through every element in the add array
+    attributeArray.forEach(element => {
+      if (AUDIENCE_LIST.includes(audienceType)) {
+        listType = audienceType;
+      }
+      if (!listType) {
+        throw new CustomError(`${audienceType} not provided`, 400);
+      }
+      switch (audienceType) {
+        case "email":
+          if (hashRequired) {
+            seedList.push(sha256(element.email));
+          } else {
+            seedList.push(element.eamil);
+          }
+          break;
+        case "deviceId":
+          if (hashRequired) {
+            seedList.push(sha256(element.deviceId));
+          } else {
+            seedList.push(element.deviceId);
+          }
+          break;
+        case "ipAddress":
+          if (hashRequired) {
+            seedList.push(sha256(element.ipAddress));
+          } else {
+            seedList.push(element.ipAddress);
+          }
+          break;
+        default:
+          throw new CustomError(
+            `Audience Type "${audienceType}" is not supported`,
+            400
+          );
+      }
+    });
+  }
+  return seedList;
+};
+
+const responseBuilder = (message, destination) => {
+  const { listData } = message.properties;
+  const { accountId, audienceId, audienceType } = destination.Config;
+
+  let outputPayload = {};
+  const typeOfOperation = Object.keys(listData);
+  typeOfOperation.forEach(key => {
```

Comment:
We don't need to loop key, value of listData its an object from which we need to take the value of `add` key

Reply:
updated